src/usb/uf2/ghostfat.c

```diff
@@ -200,26 +272,49 @@ void read_block(uint32_t block_no, uint8_t *data) {
             if (UF2_FIRST_SECTOR <= v && v <= UF2_LAST_SECTOR)
                 ((uint16_t *)(void *)data)[i] = v == UF2_LAST_SECTOR ? 0xffff : v + 1;
         }
-    } else if (block_no < START_CLUSTERS) { // Requested root directory sector
+    } else if (block_no < START_CLUSTERS) {
+
         sectionIdx -= START_ROOTDIR;
-        if (sectionIdx == 0) { // only one sector of directory entries generated
-            DirEntry *d = (void *)data;
+
+        DirEntry *d = (void *)data;
+        int remainingEntries = DIRENTRIES_PER_SECTOR;
+        if (sectionIdx == 0) { // volume label first
+            // volume label is first directory entry
             padded_memcpy(d->name, (char const *) BootBlock.VolumeLabel, 11);
             d->attrs = 0x28;
-            for (int i = 0; i < NUM_INFO; ++i) {
-                d++;
-                struct TextFile const *inf = &info[i];
-                d->size = inf->content ? strlen(inf->content) : UF2_SIZE;
-                d->startCluster = i + 2;
-                padded_memcpy(d->name, inf->name, 11);
-			}
-		}
-    } else { // else Generate the UF2 file data on-the-fly
+            d++;
+            remainingEntries--;
+        }
+
+        for (int i = DIRENTRIES_PER_SECTOR * sectionIdx;
+             remainingEntries > 0 && i < NUM_FILES;
+             i++, d++) {
+
+            // WARNING -- code presumes all but last file take exactly one sector
+            uint16_t startCluster = i + 2;
+
+            struct TextFile const * inf = &info[i];
+            padded_memcpy(d->name, inf->name, 11);
+            d->createTimeFine   = __SECONDS_INT__ % 2 * 100;
+            d->createTime       = __DOSTIME__;
+            d->createDate       = __DOSDATE__;
+            d->lastAccessDate   = __DOSDATE__;
+            d->highStartCluster = startCluster >> 8;
+            // DIR_WrtTime and DIR_WrtDate must be supported
+            d->updateTime       = __DOSTIME__;
+            d->updateDate       = __DOSDATE__;
+            d->startCluster     = startCluster & 0xFF;
+            // WARNING -- code presumes only one NULL .content for .UF2 file
+            //            and requires it be the last element of the array
+            d->size = inf->content ? strlen(inf->content) : UF2_SIZE;
+        }
+
+    } else {
```

Comment:
`// else Generate the UF2 file data on-the-fly`
I really like your comment that further explain the code in previous PR , why would you want to undo it in PR. I can add it back, no problems at all. Just wonder your reason

Reply:
I try to keep each patch branch focused on stand-alone changes.  This patch branch has not merged those prior changes.  I will work to merge them now.

Reply:
ok, manually added those changes.   not sure why they got lost, I thought GIT would be smart enough to merge them properly.